server/lib/cache.js

```diff
@@ -1,32 +1,113 @@
+import md5 from 'md5';
+import config from 'config';
+import debug from 'debug';
 import LRU from 'lru-cache';
+import Memcached from 'memcached-promisify';
+import { get, has } from 'lodash';
+
 import models from '../models';
-import debug from 'debug';
 
-const cache = LRU({
-  max: 1000,
-  maxAge: 1000 * 60 * 60 * 24, // we keep it max 1d
-});
+const debugCache = debug('cache');
 
-/** Reset cache mostly for tests */
-export function clearCache() {
-  cache.reset();
+const lru = LRU({ max: 1000 });
+
+const oneDayInSeconds = 60 * 60 * 24;
+
+let memcached;
+if (has(config, 'memcached.locations')) {
+  memcached = new Memcached(get(config, 'memcached.locations'));
 }
 
-export async function fetchCollectiveId(collectiveSlug) {
-  const collectiveId = cache.get(collectiveSlug);
-  if (collectiveId) return collectiveId;
+async function cacheGet(key) {
+  debugCache(`get ${key}`);
+  if (memcached) {
+    return memcached.get(key);
+  } else if (lru) {
+    return lru.get(key);
+  }
+}
+
+async function cacheSet(key, value, maxAgeInSeconds = 0) {
+  debugCache(`set ${key} ${value}`);
+  if (memcached) {
+    return memcached.set(key, value, maxAgeInSeconds);
+  } else if (lru) {
+    return lru.set(key, value, maxAgeInSeconds * 1000);
+  }
+}
+
+async function cacheDel(key) {
+  debugCache(`del ${key}`);
+  if (memcached) {
+    return memcached.del(key);
+  } else if (lru) {
+    return lru.del(key);
+  }
+}
+
+async function cacheClear() {
+  debugCache('clear');
+  if (lru) {
+    return lru.reset();
+  }
```

Comment:
Why not clearing the `memcached`cache too?

Reply:
Good, question! The `memcached-promisify` library I decide to use is not exposing the `flush` method. Sad!

We should use something else.